cron job `update-browser-versions` fails with "engine node is incompatible"

### Current behavior

The GitHub Actions workflow [.github/workflows/update-browser-versions.yml](https://github.com/cypress-io/cypress/blob/develop/.github/workflows/update-browser-versions.yml), running through a daily schedule defined by cron

```yml
    - cron: '0 8 * * *' #  every day at 8am UTC (3/4am EST/EDT)
```

fails with:

> error cypress@15.0.0: The engine "node" is incompatible with this module. Expected version ">=22.15.1". Got "20.19.4"

### Desired behavior

The workflow should be successful running against Cypress 15.

### Test code to reproduce

[.github/workflows/update-browser-versions.yml](https://github.com/cypress-io/cypress/blob/develop/.github/workflows/update-browser-versions.yml)

### Cypress Version

`15.0.0`

### Node version

`v20.19.4`

### Operating System

Ubuntu `24.04.2` LTS

### Debug Logs

```shell
Run actions/setup-node@v4
  with:
    node-version: 20
    always-auth: false
    check-latest: false
    token: ***
  env:
    CYPRESS_BOT_APP_ID: ***
    BASE_BRANCH: develop
Found in cache @ /opt/hostedtoolcache/node/20.19.4/x64
Environment details
  node: v20.19.4
  npm: 10.8.2
  yarn: 1.22.22
1s
Run CI=true yarn
yarn install v1.22.22
[1/6] Validating package.json...
error cypress@15.0.0: The engine "node" is incompatible with this module. Expected version ">=22.15.1". Got "20.19.4"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Error: Process completed with exit code 1.
```

### Other

[.github/workflows/update-browser-versions.yml](https://github.com/cypress-io/cypress/blob/develop/.github/workflows/update-browser-versions.yml) has hard-coded

https://github.com/cypress-io/cypress/blob/ed8987dea79e49eee38b0bafe567bd7c4857213a/.github/workflows/update-browser-versions.yml#L30-L33

instead of using the version

https://github.com/cypress-io/cypress/blob/ed8987dea79e49eee38b0bafe567bd7c4857213a/.node-version#L1

#### Suggestion

Change to

```yaml
      - name: Set up Node.js
        uses: actions/setup-node@v4
        with:
          node-version-file: .node-version
```

I'm not planning to submit a PR to resolve this issue, since I can only do this in a fork, and the workflow can't be tested in a fork. It's a one line change and that would be simpler for somebody from the Cypress.io team to do directly in the parent repo.

--- a/.github/workflows/update-browser-versions.yml
+++ b/.github/workflows/update-browser-versions.yml
@@ -30,7 +30,7 @@ jobs:
       - name: Set up Node.js
         uses: actions/setup-node@v4
         with:
-          node-version: 20
+          node-version-file: .node-version
       - name: install dependencies
         run: CI=true yarn
       - name: Check for new Chrome versions

